scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -461,11 +456,15 @@
 
                                       :cook-expected-state/starting
                                       (case pod-synthesized-state-modified
-                                        :missing (launch-pod compute-cluster
-                                                             cook-expected-state-dict pod-name)
+                                        :missing (if (:pod-deleted? synthesized-state)
+                                                   (do
+                                                     (log/info "In compute cluster" name ", pod" pod-name
+                                                               "was deleted while it was expected starting")
+                                                     (handle-pod-killed compute-cluster pod-name))
```

Comment:
Need to update the state diagram to capture this.

Reply:
Scratch that, the state diagram was already capturing this due to failed pod submissions:

```
        // (Starting, missing) -> Completed happens for some failed pod submissions
        Starting -> Completed [label=":succeeded\n:failed\n:unknown\n:missing"]
```